pkg/test/resourcefixture/test_runner.go

```diff
@@ -17,22 +17,46 @@ package resourcefixture
 import (
 	"context"
 	"fmt"
+	"strings"
 	"testing"
 
 	"github.com/GoogleCloudPlatform/k8s-config-connector/pkg/test"
 	testconstants "github.com/GoogleCloudPlatform/k8s-config-connector/pkg/test/constants"
+	"k8s.io/apimachinery/pkg/util/sets"
 )
 
 type ShouldRunFunc func(fixture ResourceFixture) bool
 type TestCaseFunc func(ctx context.Context, t *testing.T, fixture ResourceFixture)
 
 func RunTests(ctx context.Context, t *testing.T, shouldRun ShouldRunFunc, testCaseFunc TestCaseFunc) {
 	testCases := Load(t)
+
+	var filtered []ResourceFixture
 	for _, tc := range testCases {
 		if !shouldRun(tc) {
 			continue
 		}
-		runTestCase(ctx, t, tc, testCaseFunc)
+		filtered = append(filtered, tc)
+	}
+
+	// Run tests grouped by the group of the GVK
+	groups := sets.NewString()
+	for _, tc := range filtered {
+		groups.Insert(tc.GVK.Group)
+	}
+
+	for _, group := range groups.List() {
+		group := group
+		groupTestName := strings.TrimSuffix(group, ".cnrm.cloud.google.com")
+		t.Run(groupTestName, func(t *testing.T) {
+			t.Parallel()
+			for _, tc := range filtered {
+				if tc.GVK.Group != group {
```

Comment:
```suggestion
			t.Parallel()
			for _, tc := range filtered {
			        tc := tc
				if tc.GVK.Group != group {
```

do we need to make sure tc doesn't change from for loop iteration to iteration?